x/superfluid/keeper/stake.go

```diff
@@ -464,10 +473,10 @@ func (k Keeper) alreadySuperfluidStaking(ctx sdk.Context, lockID uint64) bool {
 }
 
 // mintOsmoTokensAndDelegate mints osmoAmount of OSMO tokens, and immediately delegate them to validator on behalf of intermediary account.
-func (k Keeper) mintOsmoTokensAndDelegate(ctx sdk.Context, osmoAmount osmomath.Int, intermediaryAccount types.SuperfluidIntermediaryAccount) error {
+func (k Keeper) mintOsmoTokensAndDelegate(ctx sdk.Context, osmoAmount osmomath.Int, intermediaryAccount types.SuperfluidIntermediaryAccount) (newShares math.LegacyDec, err error) {
 	validator, err := k.validateValAddrForDelegate(ctx, intermediaryAccount.ValAddr)
 	if err != nil {
-		return err
+		return math.LegacyDec{}, err
 	}
 
 	err = osmoutils.ApplyFuncIfNoError(ctx, func(cacheCtx sdk.Context) error {
```

Comment:
Semi unrelated to this PR but, do you have any idea why we are calling apply func if no error, returning the err, and then acting off that err? Imagine this was just a naked function call and we errored. Since we are acting off the error, wouldn't this just get unwound anyway?